armoryengine/Transaction.py

```diff
@@ -430,6 +431,8 @@ def TxInExtractAddrStrIfAvail(txinObj):
       return hash160_to_addrStr( hash160(lastPush) )
    elif scrType == CPP_TXIN_SPENDP2SH:
       return binScript_to_p2shAddrStr(lastPush)
+   elif scrType == CPP_TXIN_P2WPKH_P2SH or scrType == CPP_TXIN_P2WSH_P2SH:
+      return binScript_to_p2shAddrStr(hash160(rawScript))
    else:
```

Comment:
I suppose so. We can do this for the wallet later. I don't think it is necessary to do now. Or if you want to change it, go ahead (also because I don't quite know what exactly needs to be changed).